src/openms/source/ANALYSIS/XLMS/OPXLHelper.cpp

```diff
@@ -1070,7 +1052,64 @@ namespace OpenMS
         {
           ph_alpha.setMetaValue("XL_Protein_position_beta", "-");
         }
+      }
+    }
+  }
+
+  void OPXLHelper::addBetaAccessions(std::vector< PeptideIdentification > & peptide_ids)
+  {
+    for (PeptideIdentification& id : peptide_ids)
+    {
+      PeptideHit& ph_alpha = id.getHits()[0];
+
+      if (id.getHits().size() == 2)
+      {
+        PeptideHit& ph_beta = id.getHits()[1];
+        String prot2_accessions;
+
+        const std::vector<PeptideEvidence> pevs_beta = ph_beta.getPeptideEvidences();
+        for (std::vector<PeptideEvidence>::const_iterator pev = pevs_beta.begin(); pev != pevs_beta.end(); ++pev)
+        {
+          prot2_accessions = prot2_accessions + "," + pev->getProteinAccession();
+        }
 
+        prot2_accessions = prot2_accessions.suffix(prot2_accessions.size()-1);
+        ph_alpha.setMetaValue("accessions_beta", prot2_accessions);
+        ph_beta.setMetaValue("accessions_beta", prot2_accessions);
+      }
+      else
+      {
+        ph_alpha.setMetaValue("accessions_beta", "-");
+      }
+    }
+  }
+
+  void OPXLHelper::addXLTargetDecoyMV(std::vector< PeptideIdentification > & peptide_ids)
+  {
+    for (PeptideIdentification& id : peptide_ids)
+    {
+      PeptideHit& ph_alpha = id.getHits()[0];
```

Comment:
what happens if  its empty?

```suggestion
      if (id.getHits().empty()) continue;
      PeptideHit& ph_alpha = id.getHits()[0];
```